nservicebus/azure-service-bus/error-handling.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Azure Service Bus Error Handling
+reviewed: 2016-08-29
+tags:
+- Azure
+- Cloud
+- Error Handling
+---
+
+## Native Error Handling
+
+Next to the [error handling capabilities offered by NServiceBus](/nservicebus/recoverability/configure-error-handling.md), the Azure Service Bus also provides error handling capabilities of its own. This document describes the native error handling capabilities and compensating actions should there be a need. 
+
+### Azure Service Bus SDK
+
+In a cloud environment, exceptions are not exceptional at all. Given the size and complexity of a cloud environment, connectivity and service side capacity problems are to be expected. For a large part, these problems are covered by leveraging the retry behavior on transient exceptions of the Azure Service Bus SDK, known as [`RetryPolicy`](https://azure.microsoft.com/en-us/documentation/articles/best-practices-retry-service-specific/#service-bus-retry-guidelines).
+
+The following document describes a list of [common exceptions](https://azure.microsoft.com/en-us/documentation/articles/service-bus-messaging-exceptions/) that may be logged by NServiceBus. Usually, these exceptions are transient in their nature and will be automatically resolved by the retry policy of the Azure Service Bus SDK, or they are resolved at the transport level by reestablishing the connection to the service.
+
+If the exceptions persist, they will eventually trigger the transport's circuit breaker which results in crashing the process with a fatal exception.
+
+#### Broker side exceptions
+
+A subset of these exceptions originates from within the Azure Service Bus service itself. These exceptions can be recognized by the `TrackingId` field and its value in the exception message. Usually, these exceptions are also transient in nature and will disappear after a few minutes. 
+
+Note: If broker side exceptions persist the `TrackingId` value can be used to contact Azure Support to investigate further with Microsoft and Azure Service Bus teams.
+
+#### Message Size problems
+
+A peculiar behavior of the Azure Service Bus SDK is how it reports on actual message size. It can only do so accurately after a message has been sent. Before sending the reported message size only covers the body section and not the final size that would include header and serialization overhead. This can obviously lead to unexpected results when trying to send a message.
+
+If the application is intended to send large messages, it should always leverage the [NServiceBus Databus](/nservicebus/messaging/databus/) to send large payloads. In case the application sends regular messages that borderline with the maximum message size than extra precaution may be necessary.
```

Comment:
> In case the application sends regular messages that borderline with the maximum message size than extra precaution may be necessary.

how about 

> In the scenario where  an application sends regular messages that borderline with the maximum message size than extra precaution may be necessary.